### DISCUSSION TITLE
Refreshing the product list

### DISCUSSION THREAD

**Discussion by dustin** [timdorr/tesla-api#364] [Q&A] [answered]
I've recently started using the product list API (`/api/1/products`) to get the list of products and some basic state.  I'm doing two things with this:

1. Seeing the state of my car before polling deeper.
2. Getting approximate state of powerwalls.

I'm not sure about the first item since asking keeps it awake, but in the second case, I'm seeing stale data for long periods of time (e.g., the most recent poll shows me at about 89% when I'm closer to 85% -- it can stay like this for a while).

Is there a mechanism to make sure this stuff is fresh?  I saw an update after reauthenticating earlier, but I'd rather not reauthenticate frequently.

**Comment by mloyd** [comment, marked as answer]
In my experience, calling `/api/1/vehicles` keeps it awake too.  But it won't _'wake'_ it if it's asleep.  It may be different with `/api/1/products` but the result is exactly the same so I wouldn't be surprised if it behaved the same way.

Unless you have another objective reason to know your car should stay awake (e.g. sentry, charging, etc), it will take 10-15 minutes before it goes back asleep.  At which point, you can query its status without waking it.

I use a typical network exponential backoff approach when the car is not expected to be online.  That allows it to go asleep and then you can query as needed.
